src/inet/TCPEndPoint.cpp

```diff
@@ -296,6 +296,263 @@ INET_ERROR TCPEndPoint::Listen(uint16_t backlog)
     return res;
 }
 
+#if INET_CONFIG_TCP_CONN_REPAIR_SUPPORTED
+INET_ERROR TCPEndPoint::RepairConnection(const TCPConnRepairInfo &connRepairInfo, InterfaceId intf)
+{
+    INET_ERROR res = INET_NO_ERROR;
+
+#if WEAVE_SYSTEM_CONFIG_USE_LWIP
+    res = INET_ERROR_NOT_SUPPORTED;
+    ExitNow();
+#endif // WEAVE_SYSTEM_CONFIG_USE_LWIP
+
+#if WEAVE_SYSTEM_CONFIG_USE_SOCKETS
+
+#ifndef TCP_REPAIR
```

Comment:
Would it make sense to collapse all conditional compilation into the condition here?  I would certainly argue for that all of the ifdefs below should be combined here.